File: tech-studio-projects/financial-industry/algo-trading-platform/java/algo-trading-platform/src/com/fintech/algotrading/orders/RejectionReason.java
package com.fintech.algotrading.orders;

import java.util.Objects;

public class RejectionReason {
    private final String reason;

    // Constructor initializes the rejection reason.
    public RejectionReason(String reason) {
        this.reason = reason;
    }

    // Getter for the reason.
    public String getReason() {
        return reason;
    }

    // Comparison method.
    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;
        RejectionReason that = (RejectionReason) obj;
        return Objects.equals(reason, that.reason);
    }

    // Override hashCode to maintain contract between equals and hashCode.
    @Override
    public int hashCode() {
        return Objects.hash(reason);
    }
}

File: tech-studio-projects/financial-industry/algo-trading-platform/cpp/orders/RejectionReason.cpp
#include "RejectionReason.h"

// Constructor initializes the rejection reason.
RejectionReason::RejectionReason(const std::string& reason)
    : _reason(reason) {}

// Getter for the reason.
std::string RejectionReason::GetReason() const {
    return _reason;
}

// Comparison operator.
bool RejectionReason::operator==(const RejectionReason& other) const {
    return _reason == other._reason;
}
